tests/belongs_to_test.js

```diff
@@ -0,0 +1,1052 @@
+          Task = this.sequelize.define('Task', {});
+
+        Task.User = Task.belongsTo(User, { as: 'user' });
+
+        await this.sequelize.sync({ force: true });
+
+        const tasks = await Promise.all([Task.create({
+          id: 1,
+          user: { id: 1 }
+        }, {
+          include: [Task.User]
+        }), Task.create({
+          id: 2,
+          user: { id: 2 }
+        }, {
+          include: [Task.User]
+        }), Task.create({
+          id: 3
+        })]);
+
+        const result = await Task.User.get(tasks);
+        expect(result[tasks[0].id].id).to.equal(tasks[0].user.id);
+        expect(result[tasks[1].id].id).to.equal(tasks[1].user.id);
+        expect(result[tasks[2].id]).to.be.undefined;
+      });
+    });
+  });
+
+  describe('getAssociation', () => {
+    // Reason: CockroachDB guarantees that while a transaction is pending, it is isolated from other concurrent transactions with serializable isolation.
+    // https://www.cockroachlabs.com/docs/stable/transactions.html
+    it.skip('supports transactions', async function() {
+      const sequelize = this.sequelize;
+      const User = sequelize.define('User', { username: Sequelize.STRING }),
+        Group = sequelize.define('Group', { name: Sequelize.STRING });
+
+      Group.belongsTo(User);
+
+      await sequelize.sync({ force: true });
+      const user = await User.create({ username: 'foo' });
+      const group = await Group.create({ name: 'bar' });
+      const t = await sequelize.transaction();
+      await group.setUser(user, { transaction: t });
+      const groups = await Group.findAll();
+      const associatedUser = await groups[0].getUser();
+      expect(associatedUser).to.be.null;
+      const groups0 = await Group.findAll({ transaction: t });
+      const associatedUser0 = await groups0[0].getUser({ transaction: t });
+      expect(associatedUser0).to.be.not.null;
+      await t.rollback();
+    });
+
+    it('should be able to handle a where object that\'s a first class citizen.', async function() {
+      const User = this.sequelize.define('UserXYZ', { username: Sequelize.STRING, gender: Sequelize.STRING }),
+        Task = this.sequelize.define('TaskXYZ', { title: Sequelize.STRING, status: Sequelize.STRING });
+
+      Task.belongsTo(User);
+
+      await User.sync({ force: true });
+      // Can't use Promise.all cause of foreign key references
+      await Task.sync({ force: true });
+
+      const [userA, , task] = await Promise.all([
+        User.create({ username: 'foo', gender: 'male' }),
+        User.create({ username: 'bar', gender: 'female' }),
+        Task.create({ title: 'task', status: 'inactive' })
+      ]);
+
+      await task.setUserXYZ(userA);
+      const user = await task.getUserXYZ({ where: { gender: 'female' } });
+      expect(user).to.be.null;
+    });
+
+    // Reason: this fails when running on CI but works on CI on the copied tests from sequelize integration tests
```

Comment:
By copied tests, does this mean it passes when running against the sequelize test suite?

Reply:
yes

Reply:
Is there a deeper reason why this specific test fails but running against the Sequelize test suite passes? Did we override some test behaviour causing this to fail?

Reply:
Yes, that's probably it. But don't know what exactly it is.

Reply:
It was the tests that were interfering with each other. [This PR solved the issue](https://github.com/cockroachdb/sequelize-cockroachdb/pull/65)